jobsdb/jobsdb.go

```diff
@@ -1965,20 +1975,14 @@ type cacheEntry struct {
  */
 
 func (jd *HandleT) markClearEmptyResult(ds dataSetT, customer string, stateFilters []string, customValFilters []string, parameterFilters []ParameterFilterT, value cacheValue, checkAndSet *cacheValue) {
-	if !jd.Multitenant {
-		customer = undefinedWorkspace
-	} else if customer == undefinedWorkspace {
-		panic(fmt.Errorf("%s: undefined workspace is not allowed for multitenant", jd.tablePrefix))
-	}
-
 	jd.dsCacheLock.Lock()
 	defer jd.dsCacheLock.Unlock()
 
 	//This means we want to mark/clear all customVals and stateFilters
 	//When clearing, we remove the entire dataset entry. Not a big issue
 	//We process ALL only during internal migration and caching empty
 	//results is not important
-	if len(stateFilters) == 0 || len(customValFilters) == 0 || customer == `` {
+	if len(stateFilters) == 0 || len(customValFilters) == 0 {
```

Comment:
Why we no longer check for `customer == '' `

Reply:
Ideally we should never receive customer as empty string.